### PR TITLE
Implement CloudWatch get_metric_statistics

### PR DESCRIPTION
**Pull request by thehesiod**
- implements CloudWatch `get_metric_statistics`
- refactors `_get_multi_param` to support recursive lists of dictionaries

### CODE REVIEW

File: moto/cloudwatch/models.py [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
 from moto.core import BaseBackend, BaseModel
 from moto.core.exceptions import RESTError
 import boto.ec2.cloudwatch
-import datetime
-
+from datetime import datetime, timedelta
+from dateutil.tz import tzutc
 from .utils import make_arn_for_dashboard
-
+import statistics
```
Comment (terrycain) [inline moto/cloudwatch/models.py]: Is this available on Python2? and if not, its only used for mean, couldnt that be replaced with sum() / len()

**Comment by terrycain** [comment]
Looks like some really good improvements, but flake8 isnt too happy :wink:

File: moto/cloudwatch/models.py (hunk 1 above)
Comment (thehesiod) [reply moto/cloudwatch/models.py]: new in 3.4 :(  I'll update it

Comment (terrycain) [reply moto/cloudwatch/models.py]: Yeah I know how you feel, shame moto can't be py35+ only

**Comment by thehesiod** [comment]
pycharm didn't remove the whitespace since it was in a giant comment, good, but dumb on part of flake8

File: moto/cloudwatch/models.py [hunk 2]
```diff
@@ -92,6 +121,53 @@ def __repr__(self):
...
+            return None
+
+        return max(self.values)
+
+    @property
+    def average(self):
+        if 'Average' not in self.stats:
+            return None
+
+        # when moto is 3.4+ we can switch to the statistics module
```
Comment (terrycain) [inline moto/cloudwatch/models.py]: :+1:

**Comment by terrycain** [comment]
Other than the tests not passing, all the code looks good :+1: will merge once travis is happy

**Comment by thehesiod** [comment]
wish this was 3.6+ so I could add all the typing I wanted :)

**Comment by terrycain** [comment]
typing ftw

**Comment by thehesiod** [comment]
https://www.youtube.com/watch?v=qCRae5mRoRE

**Comment by terrycain** [comment]
Haha

**Comment by thehesiod** [comment]
TY!

**Comment by thehesiod** [comment]
@JackDanger just want to mention enhancement to `_get_multi_param_helper`, it's a lot easier to support recursive lists of dicts

**Pull request merged**